auth/role_info.hh

```diff
@@ -36,4 +45,65 @@ struct role_config_update final {
     std::optional<bool> can_login{};
 };
 
+struct role_info : public data_dictionary::keyspace_element {
+    sstring role_name;
+    role_config config;
+    std::optional<sstring> maybe_salted_hash;
+
+    virtual ~role_info() = default;
+
+    virtual sstring keyspace_name() const override {
+        return "<null>";
+    }
```

Comment:
I must've misunderstood the design doc. I was convinced we were supposed to return this very string. The name of a keyspace cannot contain `<` or `>`, so there wasn't any ambiguity. I'll change it to an empty string, though I'm afraid it will be turned into two quotation marks if I don't change some portion of the existing code.